client/gutenberg/editor/controller.js

```diff
@@ -216,6 +216,15 @@ function getAnchorFmData( query ) {
 	return { anchor_podcast, anchor_episode, spotify_url };
 }
 
+function showDraftPostModal() {
+	const value = window.sessionStorage.getItem( 'wpcom_signup_complete_show_draft_post_modal' );
+	if ( value ) {
+		window.sessionStorage.removeItem( 'wpcom_signup_complete_show_draft_post_modal' );
+	}
+
+	return value;
+}
```

Comment:
Fair enough wanting to keep the user url looking clean. Users should also be able to refresh the page without seeing seeing the modal again.

This function isn't called within a React render so repeated calls are unlikely, but it still feels like it should be safe to call this function more than once.

```suggestion
let showDraftPostModalValue = 'init';
function showDraftPostModal() {
	if ( showDraftPostModalValue === 'init' ) {
		showDraftPostModalValue = window.sessionStorage.getItem( 'wpcom_signup_complete_show_draft_post_modal' 
		window.sessionStorage.removeItem( 'wpcom_signup_complete_show_draft_post_modal' );
	}

	return showDraftPostModalValue;
}
```

Reply:
I'm not sure we can save the value or not. If this function might be called multiple times, we could not know it's the first time or not so that the modal might show again.

Reply:
That's actually a really good point! We only ever want the modal to appear once, so it's actually useful to have the value "self-destruct" so to speak. I stand corrected 🙇‍♂️